src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2293,12 +2293,145 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
+                        return;
+
                     default:
                         return;
                 }
             }
         }
 
+        // All conflict markers consist of the same character repeated seven times.  If it is
+        // a <<<<<<< or >>>>>>> marker then it is also followed by a space.
+        private static readonly int s_conflictMarkerLength = "<<<<<<<".Length;
+
+        private bool IsConflictMarkerTrivia()
+        {
+            var position = TextWindow.Position;
+            var text = TextWindow.Text;
+            if (position == 0 || SyntaxFacts.IsNewLine(text[position - 1]))
+            {
+                var firstCh = text[position];
+                Debug.Assert(firstCh == '<' || firstCh == '=' || firstCh == '>');
+
+                if ((position + s_conflictMarkerLength) <= text.Length)
+                {
+                    for (int i = 0, n = s_conflictMarkerLength; i < n; i++)
+                    {
+                        if (text[position + i] != firstCh)
+                        {
+                            return false;
+                        }
+                    }
+
+                    if (firstCh == '=')
+                    {
+                        return true;
+                    }
+
+                    return (position + s_conflictMarkerLength) < text.Length &&
+                        text[position + s_conflictMarkerLength] == ' ';
+                }
+            }
+
+            return false;
+        }
+
+        private void LexConflictMarkerTrivia(ref SyntaxListBuilder triviaList)
+        {
+            this.AddError(TextWindow.Position, s_conflictMarkerLength,
+                ErrorCode.ERR_Merge_conflict_marker_encountered);
+
+            var startCh = this.TextWindow.PeekChar();
+
+            // First create a trivia from the start of this merge conflict marker to the
+            // end of line/file (whichever comes first).
+            LexConflictMarkerHeader(ref triviaList);
+
+            // Now add the newlines as the next trivia.
+            LexConflictMarkerEndOfLine(ref triviaList);
+
+            // Now, if it was an ======= marker, then also created a DisabledText trivia for
+            // the contents of the file after it, up until the next >>>>>>> marker we see.
+            if (startCh == '=')
+            {
+                LexConflictMarkerDisabledText(ref triviaList);
+            }
+        }
+
+        private SyntaxListBuilder LexConflictMarkerDisabledText(ref SyntaxListBuilder triviaList)
+        {
+            // Consume everything from the start of the mid-conflict marker to the start of the next
+            // end-conflict marker.
+            this.Start();
+
+            while (true)
+            {
+                var ch = this.TextWindow.PeekChar();
+                if (ch == SlidingTextWindow.InvalidCharacter)
+                {
+                    break;
+                }
+
+                if (ch == '>' && IsConflictMarkerTrivia())
```

Comment:
IsConflictMarkerTrivia actually verifies that it's at the start of a line.  Which is pretty much all that we want since htat's what you always get from Git.